alembic/versions/20231101_2d72d6876c52_remove_collection_external_integration.py

```diff
@@ -0,0 +1,273 @@
+"""Remove collection external integration.
+
+Revision ID: 2d72d6876c52
+Revises: cc084e35e037
+Create Date: 2023-11-01 22:42:06.754873+00:00
+
+"""
+import logging
+
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+from alembic import op
+from api.integration.registry.license_providers import LicenseProvidersRegistry
+from core.model import json_serializer
+
+# revision identifiers, used by Alembic.
+revision = "2d72d6876c52"
+down_revision = "cc084e35e037"
+branch_labels = None
+depends_on = None
+
+
+log = logging.getLogger(f"palace.migration.{revision}")
+log.setLevel(logging.INFO)
+log.disabled = False
+
+
+def upgrade() -> None:
+    conn = op.get_bind()
+
+    # Our collection names have gotten out of sync with the integration names. The collection names
+    # are what are being displayed to users, so before we stop using the collection name, we need
+    # to update the integration name to match the collection name.
+    # For now, we leave the collection name column in place, but we make it nullable and remove the
+    # unique constraint.
+    rows = conn.execute(
+        "SELECT c.id as collection_id, ic.id as integration_id, ic.name as integration_name, "
+        "c.name as collection_name from collections c JOIN integration_configurations ic "
+        "ON c.integration_configuration_id = ic.id WHERE c.name != ic.name"
+    ).all()
+
+    for row in rows:
+        log.info(
+            f"Updating name for collection {row.collection_id} from {row.integration_name} to {row.collection_name}."
+        )
+        conn.execute(
+            "UPDATE integration_configurations SET name = (%s) WHERE id = (%s)",
+            (row.collection_name, row.integration_id),
+        )
+
+    op.alter_column("collections", "name", existing_type=sa.VARCHAR(), nullable=True)
+    op.drop_index("ix_collections_name", table_name="collections")
+
+    # We have moved the setting for the TOKEN_AUTH integration from an external integration
+    # to a new JSONB column on the integration_configurations table (context). We need to move
+    # the data into the new column as part of this migration.
+    # The context column is not nullable, so we need to set a default value for the existing
+    # rows. We will use an empty JSON object. We create the column as nullable, set the default
+    # value, then make it non-nullable.
+    op.add_column(
+        "integration_configurations",
+        sa.Column("context", postgresql.JSONB(astext_type=sa.Text()), nullable=True),
```

Comment:
Ah ok, yes `default` is an application level default. Using `server_default` will work, I tested it out.

Reply:
I think this changes the DB level schema though, right?

So migrated tables would have a different schema definition then tables that weren't migrated. We could mitigate this by changing the column definition to use `server_default`, but since none of our other tables with defaults use `server_default`, I'd rather leave it as `default` for consistency.